channeldb/payments.go

```diff
@@ -298,42 +298,8 @@ func fetchPayment(bucket kvdb.RBucket) (*MPPayment, error) {
 		failureReason = &reason
 	}
 
-	// Go through all HTLCs for this payment, noting whether we have any
-	// settled HTLC, and any still in-flight.
-	var inflight, settled bool
-	for _, h := range htlcs {
-		if h.Failure != nil {
-			continue
-		}
-
-		if h.Settle != nil {
-			settled = true
-			continue
-		}
-
-		// If any of the HTLCs are not failed nor settled, we
-		// still have inflight HTLCs.
-		inflight = true
-	}
-
-	// Use the DB state to determine the status of the payment.
-	var paymentStatus PaymentStatus
-
-	switch {
-	// If any of the the HTLCs did succeed and there are no HTLCs in
-	// flight, the payment succeeded.
-	case !inflight && settled:
-		paymentStatus = StatusSucceeded
-
-	// If we have no in-flight HTLCs, and the payment failure is set, the
-	// payment is considered failed.
-	case !inflight && failureReason != nil:
-		paymentStatus = StatusFailed
-
-	// Otherwise it is still in flight.
-	default:
-		paymentStatus = StatusInFlight
-	}
+	// Now determine the payment's status.
+	paymentStatus := decidePaymentStatus(htlcs, failureReason)
```

Comment:
Would squash this change with the commit where `decidePaymentStatus` is introduced, so that one can see the difference in one view.

Reply:
I'd rather not as the two commits are already big. Maybe those can be rearranged to use three commits, introduce `decidePaymentStatus` and apply, add unit test, and fix unit test? Anyway updated the commit message to address the difference.

Reply:
I am just commenting on this from a reviewer perspective. Needing to flip back and forth between commits isn't ideal. Reviewing a commit that basically contains dead code is difficult because there is no reference frame. It would be different if it would be some kind of isolated implementation of a well known algorithm for example. But in this case, the `decidePaymentStatus` is very tightly coupled to the rest of the code.

Reply:
sure yeah, I'm all about reviewer-oriented programming. I usually put the unit tests and the function tested in the same commit, so it's easier to check if all logical branches are hit. I also thought we were building something very new here, so it's better to forget how we handle statuses previously and check whether the newly added methods make sense or not. If not, we can question it in this commit, till the point we accept that it's initable, registrable, etc. Of course comparing them can be nice, it's just the tradeoff I made when limiting diff sizes. Now it's switched again, unit tests are removed from the commit and its callsites are added back.